### PR TITLE
Add CRUD endpoints for Book model

### PR DESCRIPTION
**Pull request by martinremis**
## Summary
* Book crud endpoints creation
* Add custom exceptions
## Trello card
https://trello.com/c/XSCroj0O/18-crear-las-operaciones-crud-de-book

### CODE REVIEW

File: src/main/java/wolox/training/controllers/BookController.java [hunk 1]
```diff
@@ -20,4 +32,34 @@ public String greeting(@RequestParam(name="name", required=false, defaultValue="
     return "greeting";
   }
 
+  @GetMapping("/{id}")
+  public Optional<Book> read(@PathVariable Long id) {
+    Optional<Book> book = bookRepository.findById(id);
```
Comment (rodrigofranc) [inline src/main/java/wolox/training/controllers/BookController.java]: Check `orElseThrow ` method from optionals

File: src/main/java/wolox/training/exceptions/NotFound.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package wolox.training.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+public class NotFound extends ResponseStatusException  {
```
Comment (rodrigofranc) [inline src/main/java/wolox/training/exceptions/NotFound.java]: Always finish the class with `Exception` in this case could be `BookNotFoundException `

**Review by rodrigofranc — CHANGES_REQUESTED** [review]

**Review by rabinovichm — APPROVED** [review]
Nice work :blush:

File: src/main/java/wolox/training/controllers/BookController.java [hunk 3]
```diff
@@ -20,4 +33,28 @@ public String greeting(@RequestParam(name="name", required=false, defaultValue="
...
+  @DeleteMapping("/{id}")
+  public void delete(@PathVariable Long id) {
+    bookRepository.findById(id).orElseThrow(() -> new NotFoundException(notFoundMessage));
+    bookRepository.deleteById(id);
+  }
+
+  @PutMapping("/{id}")
+  public Book update(@RequestBody Book book, @PathVariable Long id) {
+    bookRepository.findById(id).orElseThrow(() -> new NotFoundException(notFoundMessage));
+    if (!book.getId().equals(id)) throw new ForbiddenException( "Cannot change 'id'");
```
Comment (rodrigofranc) [inline src/main/java/wolox/training/controllers/BookController.java]: It is good practice to have all error messages as constants in a separate class, we should not have them literal in the place where the exception is raised

**Review by rodrigofranc — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/controllers/BookController.java (hunk 3 above)
Comment (martinremis) [reply src/main/java/wolox/training/controllers/BookController.java]: Where should the class be? Under the main package?

Comment (rodrigofranc) [reply src/main/java/wolox/training/controllers/BookController.java]: yes, and under a package named `constants` for more readability

File: src/main/java/wolox/training/constants/ErrorMessages.java [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package wolox.training.constants;
+
+public class ErrorMessages {
+  public static String notFoundMessage = "Book not found";
```
Comment (rodrigofranc) [inline src/main/java/wolox/training/constants/ErrorMessages.java]: This should be declared as constants

**Review by rodrigofranc — APPROVED** [review]

**Pull request merged**